Several changes to blocks

SquidDev opened the pull request:
Apologies for PR spam, just this one has a couple of changes I was unsure of, so it felt cleaner to split it into something separate:

 - I'm a bit conflicted on `.getCollisionShape` - it appears to be used both
for entity collision and determining if a block's side is a "full cube" (and so can have torches placed against or w/e)
 - I've renamed `getRenderingState` to `getStateForNeighborUpdate`, though am not 100% it's a good name - it may also perform side effects (mostly scheduling this block to be ticked next thread), so isn't quite a pure getter.

SquidDev committed:
Some more mappings for block shapes

I'm a bit conflicted on .getCollisionShape - it appears to be used both
for entity collision and determining if a side is solid.

SquidDev committed:
getRenderingState -> getStateForNeighborUpdate

This is no longer used just for rendering, and may perform all sorts of
side effects: such as scheduling a water update.

gegy1000 commented:
`getCollisionShape` could possibly be something along the lines of `getSolidShape`?

shadowfacts commented on mappings/net/minecraft/block/Block.mapping:
```diff
@@ -450,7 +450,7 @@ CLASS bgs net/minecraft/block/Block
 	METHOD n hasRandomTicks (Lbpm;)Z
 		ARG 1 state
-	METHOD n (Lbpm;Lbbb;Let;)Z
+	METHOD n isFullCube (Lbpm;Lbbb;Let;)Z
```
Can you clarify the difference between `isFullCube` and the two `isShapeFullCube` methods?

SquidDev replied:
`isShapeFullCube` operates on a `VoxelShape` - it basically checks if it's equivalent to `VoxelShapes.fullCube()`. `isFullCube` accepts a world, block state and block position, gets the collision box and then uses `isFullCube`.

IMO it's fairly self-explanatory when you have the mapped class names too, but happy to rename it to something more specific if people prefer :).

asiekierka replied:
I think we should go with this for now and see what happens. I'm all for naming Block methods based on *uses*, not *behavior*, but that's tricky.

SquidDev committed:
isPistonExtension -> hasDynamicBounds

This is used on both instances where the bounding or collision box of a
block is dynamic. Namely, it is based on the state of the tile entity
rather than the the current block state.

SquidDev commented:
Sorry for augmenting the PR: the above methods was pretty linked to the other mapping done here with bounding and collision boxes, so it felt it made sense to do it in here. Happy to cherry pick it out if preferred though.

SquidDev commented:
🎉 ✨ Thank you all for reviewing!
